featuremanagement/_time_window_filter/_models.py

```diff
@@ -0,0 +1,119 @@
+# ------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+from enum import Enum
+from typing import Dict, Any
+from datetime import datetime
+from dataclasses import dataclass
+from email.utils import parsedate_to_datetime
+
+
+class RecurrencePatternType(str, Enum):
+    """
+    The recurrence pattern type.
+    """
+
+    DAILY = "Daily"
+    WEEKLY = "Weekly"
+
+    @staticmethod
+    def from_str(value: str) -> "RecurrencePatternType":
+        """
+        Get the RecurrencePatternType from the string value.
+
+        :param value: The string value.
+        :type value: str
+        :return: The RecurrencePatternType.
+        :rtype: RecurrencePatternType
+        """
+        if value == "Daily":
+            return RecurrencePatternType.DAILY
+        if value == "Weekly":
+            return RecurrencePatternType.WEEKLY
+        raise ValueError(f"Invalid value: {value}")
+
+
+class RecurrenceRangeType(str, Enum):
+    """
+    The recurrence range type.
+    """
+
+    NO_END = "NoEnd"
+    END_DATE = "EndDate"
+    NUMBERED = "Numbered"
+
+    @staticmethod
+    def from_str(value: str) -> "RecurrenceRangeType":
+        """
+        Get the RecurrenceRangeType from the string value.
+
+        :param value: The string value.
+        :type value: str
+        :return: The RecurrenceRangeType.
+        :rtype: RecurrenceRangeType
+        """
+        if value == "NoEnd":
+            return RecurrenceRangeType.NO_END
+        if value == "EndDate":
+            return RecurrenceRangeType.END_DATE
+        if value == "Numbered":
+            return RecurrenceRangeType.NUMBERED
+        raise ValueError(f"Invalid value: {value}")
+
+
+class RecurrencePattern:  # pylint: disable=too-few-public-methods
+    """
+    The recurrence pattern settings.
+    """
+
+    def __init__(self, pattern_data: Dict[str, Any]):
+        self.type = RecurrencePatternType.from_str(pattern_data.get("Type", "Daily"))
+        self.interval = pattern_data.get("Interval", 1)
+        self.days_of_week = pattern_data.get("DaysOfWeek", [])
```

Comment:
How does it convert day of week string("Monday"）to index/number?

Reply:
Isn't it always just a number? Otherwise, how do we deal with multiple languages? De we just assume English?

Reply:
The schema is here: https://github.com/microsoft/FeatureManagement/blob/main/Schema/FeatureFilters/Microsoft.TimeWindow.v2.0.0.schema.json

We assume English. We followed outlook schema: https://learn.microsoft.com/en-us/graph/api/resources/recurrencepattern?view=graph-rest-1.0

> how do we deal with multiple languages

All of our API and schema are English.